### ISSUE TITLE
Rotation centers have hidden behavior that is confusing.

### ISSUE THREAD

**Issue by openingapps** [opentoonz/opentoonz#4033]
I have been seeing unexpected behavior in opentoonz relating to the center of rotation and position for imported objects. In particular, the x and y coordinates shown in opentoonz don't always match what is drawn in the viewer.

This has been discussed before in closed issue 1755, but I don't think it was completely answered.  I have more information than the original issue gave.  So I'm filing a new issue.

Here is the sequence of events that causes non-inutitive behavior:

Move the center of a level

Rotate the level

Move the center of the level again

Rotate the level again

As one might expect, this sequence of events effectively changes the position of the level on the screen. That is intuitive. BUT, none of the parameters displayed on the screen indicate that the position has changed.   

If you open the saved scene (tnz) file in an text editor, you will see that level object (called pegbar in the file) has a "center" property with 4 values. These 4 values implement the position change described above. But only the first 2 values are ever shown to the user.  

It might have been nice to tell the user that the position is changed for this reason.

I also posted on reddit.  There is a picture in that post:  https://www.reddit.com/r/OpenToonz/comments/oxyuja/there_is_more_to_center_position_than_you_can_see/

**Comment by RodneyBaker** [comment]
@openingapps I'm tagging this as 'needs info' although you have provided quite a lot of information.
I'm mostly adding the label so others will know to join in the discussion and provide additional feedback to help move your request forward.

**Comment by RodneyBaker** [comment]
Something I just now noticed in your image on reddit is that you are adjusting the Z-depth.
That definitely needs to be accounted for in our troubleshooting as well as any other transforms.

**Comment by openingapps** [comment]
@RodneyBaker 
In the example, I don't think the Z-depth changes the behavior that I saw.

My rendering is 100% 2D.   I think I was just using z-depth to make sure that the head of the horse was shown "in front" of the body.    Since I am not using any perspective in the camera, the z-depth does not doesn't change how the head is centered in the rendered picture.

**Comment by openingapps** [comment]
Hi Rodney,

No really, this has nothing to do with z-depth.  All of the horse parts are in the same z plane.  I just dug up my old project and verified it.  The depth of the horse's neck is 16.  The depth of the horse's head when it is still connected is 16.  The depth of the horse's disconnected head is also 16.

![Image](https://github.com/user-attachments/assets/d78582ee-1df0-44b7-b977-710b3c2d64eb)
![Image](https://github.com/user-attachments/assets/3daa0c6b-eccd-4633-b53f-ca668834f8a3)
![Image](https://github.com/user-attachments/assets/2ca02966-7939-439f-82b1-a41da6d72d8b)


But to reiterate my original point.  **There is nothing wrong with the openToonz rendering algorithm.**  If you move the horse head's center, rotate, move the center back, and rotate back -- then the horse head should be shown in a different place.  That is absolutely correct!!

**What is wrong is the user interface.**  The UI says that the head position of the disconnected is still at (0,0), and there is no indication as to why it is not actually drawn at (0,0).

There is also no way to undo the effect of the problem once the project has been saved.  There are two parameters in the database that the user can't reset.

Thanks,
openingapps

**Comment by RodneyBaker** [comment]
I'm just trying to figure out what I'm missing here.
In every image you've shared (in the last post) there is a Z-depth indicator on screen.
This appears when we use Control and move our mouse pen.
I get that the fielding says 16 on your screen so that would indicate that at least on that key/frame you are in 16 field.

Is there some reason you are using that Control key that invokes z-depth such as for moving into and out of the image?

To reiterate, the only way I know to get that Z-depth gizmo to appear is to initiate movement of z-depth via use of the control key.

None of your images show the entire screen or your keyframes so is it possible the frame you are showing is 16 fielding but other frames it is different?

Perhaps you can share the scene file via File > Export Scene and zip up the contents of that directory?

I can try to duplicate what you've got but I'm completely guessing.

https://github.com/user-attachments/assets/d6f439ee-cf01-4131-8d5e-fc1157da3245

Is this (the attached video) not similar to what you are seeing?

**Comment by openingapps** [comment]
Yeah.  That's interesting!!  I hadn't noticed the z-depth indicator, and the fact that it should not be there.

There is an explanation...

The reason you see the z-depth indicator is because the key shortcut for taking a screen shot on mac-os is cmd-shift-5.
But opentoonz has also a short-cut for adjusting z-depth that is cmd-mouse.

**As soon as I press the cmd button to take a screen shot, opentoonz displays a z-depth indicator in its GUI.  That indicator ultimately gets captured in the screen shot.  I am not adjusting the z-depth.  Opentoonz is just displaying an indicator that says that I pressed cmd, and I am taking a picture of it.**

About frames...
In this example, there is no animation and no transition between key frames.  I only have one frame, and all of the center adjusting and rotating has occured in that frame.

![Image](https://github.com/user-attachments/assets/c88f6534-8bfb-4f5c-9218-21d22cb886ad)

**Comment by openingapps** [comment]
Hi again,

Here is a video that shows the behavior.  Note that at the start and end of the video, the head's position shows in the UI as (0,0).  But the head has moved.  After you save this scene, it is too late to use "undo".  There is no indication why the head looks wrong, or any record of how the head got this way.  The head is permanently displaced and there is no way to fix it.

(You actually can fix the head by editing the database file and reseting the 2 hidden parameters.  But most people won't know how to do that.)

https://github.com/user-attachments/assets/4ecc14a6-4fdf-410d-9d6d-ac934740198c
